src/shared/bus-unit-util.c

```diff
@@ -1180,6 +1240,82 @@ static int bus_append_service_property(sd_bus_message *m, const char *field, con
 
                 return bus_append_exec_command(m, field, eq);
 
+        if (STR_IN_SET(field, "RestartPreventExitStatus", "RestartForceExitStatus", "SuccessExitStatus")) {
+                _cleanup_free_ int *status = NULL, *signal = NULL;
+                size_t sz_status = 0, sz_signal = 0;
+                const char *p;
+
+                for (p = eq;;) {
+                        _cleanup_free_ char *word = NULL;
+                        int val;
+
+                        r = extract_first_word(&p, &word, NULL, EXTRACT_QUOTES);
+                        if (r == 0)
+                                break;
+                        if (r == -ENOMEM)
+                                return log_oom();
+                        if (r < 0)
+                                return log_error_errno(r, "Invalid syntax in %s: %s", field, eq);
+
+                        r = safe_atoi(word, &val);
+                        if (r < 0) {
+                                val = signal_from_string_try_harder(word);
+                                if (val < 0)
+                                        return log_error_errno(r, "Invalid status or signal %s in %s: %m", word, field);
+
+                                signal = realloc_multiply(signal, sizeof(int), sz_signal + 1);
```

Comment:
not that it matters, but usually we use GREEDY_REALLOC() for this, to make this slightly more efficient

Reply:
Oh, I forgot that. I will update. Thank you.